### PR TITLE
Add more validations.

### PR DESCRIPTION
**Pull request by jortel**
Another round of validation work.

- Moved `MigPlan` update predicate to its own module and applied this pattern to the `MigStorage` and `MigAssetCollection` controllers.
- Defined `message` constants and added to _raised_ conditions.  Having the message seems helpful users and likely to be helpful for UI.
- Adds validations to `MigAssetCollection` controller.

### CODE REVIEW

File: pkg/controller/migassetcollection/predicate.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type UpdatedPredicate struct {
+	predicate.Funcs
+}
+
+func (r UpdatedPredicate) Update(e event.UpdateEvent) bool {
```
Comment (djwhatle) [inline pkg/controller/migassetcollection/predicate.go]: +1 on sharing this across controllers

File: pkg/controller/migplan/validation.go [hunk 2]
```diff
@@ -33,17 +33,44 @@ const (
 	False = "False"
 )
 
+// Message
+const (
+	ReadyMessage                        = "The migration plan is ready."
+	InvalidSourceClusterRefMessage      = "A valid source cluster resource has not been associated with this plan."
```
Comment (djwhatle) [inline pkg/controller/migplan/validation.go]: Are these intended for the UI? I'm wondering, from a CLI user POV, if a message explicitly mentioning the field needing attention (sourceClusterRef in this case) would be easier to rectify.

Comment (djwhatle) [reply pkg/controller/migplan/validation.go]: Could read something like "A valid source cluster reference (srcClusterRef) has not been provided with this plan."

Comment (jortel) [reply pkg/controller/migplan/validation.go]: It's mainly intended for a human that is editing yaml but thought it _may_ be used by the UI.  I considered including field name causing the problem but was a little concerned about keeping it in sync with the field _tags_ (yaml).  Also, if used by the UI, the field names would have little meaning.  By not including the field name but describing the data instead, it correlates more loosely to the yaml fileds which is less helpful but also won't get out of sync and may be more broadly useful.
It's a trade.  Not sure what the right approach is.

Comment (jortel) [reply pkg/controller/migplan/validation.go]: One possible way to mitigate the problem of keeping the field name in the API (yaml) and the field name in the condition message the same is to use _reflect_ to get the struct field tag, parse it to get the name and use it in the message.

Comment (jortel) [reply pkg/controller/migplan/validation.go]: Using reflect may not be a good idea either.  I looked into using reflect and the code would need to pass the field name (string) which could also get out of sync.

Comment (jortel) [reply pkg/controller/migplan/validation.go]: I reworded the `message` to include the json field name.  I agree this is better.

**Pull request merged**